common/src/main/java/com/hedera/block/common/utils/Preconditions.java

```diff
@@ -185,6 +185,50 @@ public static long requireGreaterOrEqual(final long toTest, final long base, fin
         throw new IllegalArgumentException(message);
     }
 
+    /**
+     * This method asserts a given int is within a range (boundaries
+     * included). If the given int is within the range, then we return it,
+     * else, an {@link IllegalArgumentException} is thrown.
+     *
+     * @param toCheck the int value to test
+     * @param lowerBoundary the lower boundary
+     * @param upperBoundary the upper boundary
+     * @return the input {@code toCheck} if it is within the range (boundaries
+     * included)
+     * @throws IllegalArgumentException if the input int does not pass the test
+     */
+    public static int requireInRange(final int toCheck, final int lowerBoundary, final int upperBoundary) {
+        return requireInRange(toCheck, lowerBoundary, upperBoundary, null);
+    }
+
+    /**
+     * This method asserts a given int is within a range (boundaries
+     * included). If the given int is within the range, then we return it,
+     * else, an {@link IllegalArgumentException} is thrown.
+     *
+     * @param toCheck the int value to check
+     * @param lowerBoundary the lower boundary
+     * @param upperBoundary the upper boundary
+     * @param errorMessage the error message to be used in the exception if the
+     * input int to test is not within the range, if null, a default message
+     * will be used
+     * @return the input {@code toCheck} if it is within the range (boundaries
+     * included)
+     * @throws IllegalArgumentException if the input int does not pass the test
+     */
+    public static int requireInRange(
+            final int toCheck, final int lowerBoundary, final int upperBoundary, final String errorMessage) {
+        if (toCheck >= lowerBoundary && toCheck <= upperBoundary) {
+            return toCheck;
+        } else {
+            final String message = Objects.isNull(errorMessage)
+                    ? "The input int [%d] is required to be in the range [%d, %d] boundaries included."
+                            .formatted(toCheck, lowerBoundary, upperBoundary)
+                    : errorMessage;
+            throw new IllegalArgumentException(message);
+        }
+    }
+
```

Comment:
Just FYI as an example.  Something to work on in _future_ PRs.
Here is how we could make `errorMessage` `@NonNull` and still be efficient:
```suggestion
    private static final String OUT_OF_RANGE_DEFAULT_MESSAGE =
            "The input value [%d] is required to be in the range [%d, %d], inclusive.";
    /**
     * This method asserts a given int is within a range (boundaries
     * included). If the given int is within the range, then we return it,
     * else, an {@link IllegalArgumentException} is thrown.
     *
     * @param toCheck the int value to test
     * @param lowerBoundary the lower boundary
     * @param upperBoundary the upper boundary
     * @return the input {@code toCheck} if it is within the range (boundaries
     * included)
     * @throws IllegalArgumentException if the input int does not pass the test
     */
    public static int requireInRange(final int toCheck, final int lowerBoundary, final int upperBoundary) {
        return requireInRange(toCheck, lowerBoundary, upperBoundary, OUT_OF_RANGE_DEFAULT_MESSAGE);
    }

    /**
     * This method asserts a given int is within a range (boundaries
     * included). If the given int is within the range, then we return it,
     * else, an {@link IllegalArgumentException} is thrown.
     *
     * @param toCheck the int value to check
     * @param lowerBoundary the lower boundary
     * @param upperBoundary the upper boundary
     * @param errorMessage A format string {@see Formatter#syntax} with three
     *     decimal parameters for {@code toCheck}, {@code lowerBoundary},
     *     and {@code upperBoundary}.<br/>
     *     Example: {@code "The input (%d) is out of range (%d, %d)"}
     * @return the input {@code toCheck} if it is within the range (boundaries
     * included)
     * @throws IllegalArgumentException if the input int does not pass the test
     */
    public static int requireInRange(
            final int toCheck, final int lowerBoundary, final int upperBoundary, final String errorMessage) {
        if (toCheck >= lowerBoundary && toCheck <= upperBoundary) {
            return toCheck;
        } else {
            final String message = errorMessage.formatted(toCheck, lowerBoundary, upperBoundary);
            throw new IllegalArgumentException(message);
        }
    }

```

Reply:
@jsync-swirlds that is indeed a good proposal. One thing though, having a pre-defined template to resolve is quite restrictive (we cannot provide anything outside the pre-defined template), so I think a better api and implementation would be the following:

```java
    public static int requireInRange(final int toCheck, final int lowerBoundary, final int upperBoundary) {
        return requireInRange(
                toCheck,
                lowerBoundary,
                upperBoundary,
                "The input value [%d] is required to be in the range [%d, %d], inclusive.",
                toCheck,
                lowerBoundary,
                upperBoundary);
    }

    public static int requireInRange(
            final int toCheck,
            final int lowerBoundary,
            final int upperBoundary,
            @NonNull final String errorMessage,
            final Object... errorMessageArgs) {
        if (toCheck >= lowerBoundary && toCheck <= upperBoundary) {
            return toCheck;
        } else {
            final String message = errorMessage.formatted(errorMessageArgs);
            throw new IllegalArgumentException(message);
        }
    }
```

This gives us complete freedom over what the message would be, while we keep the lazy formatting for the message. I think this is the best way to go, since we can have any template we chose or no template at all. So let me know also on your opinion, but I think this is the way to go.

Reply:
I don't expect to handle this in this issue; the comment was just an example for the future chore.

What I wrote doesn't set a single pre-defined template, it has a _default_ template and the option to replace that template with anything the caller prefers.  The only pre-defined element is what values are filled in.  Given that the only thing a caller cannot pre-populate before calling is the parameters; what you propose is overkill (not to mention _more_ costly as you're passing the same parameters multiple times and wrapping primitives in most cases).

It's vastly simpler, and no less flexible, to stick with the approach I suggested.